clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/main/MainIngestionMonitorService.java

```diff
@@ -72,6 +77,8 @@ public class MainIngestionMonitorService extends AbstractVeniceService {
   private long connectionTimeoutMs;
   private volatile long latestHeartbeatTimestamp = -1;
 
+  private final ReentrantReadWriteLock forkProcessLeaderStateActionLock = new ReentrantReadWriteLock();
```

Comment:
Nit: let's add some comments to explain a bit about the intention of this RW lock, if makes sense.